azurerm/internal/services/apimanagement/api_management_gateway_data_source.go

```diff
@@ -0,0 +1,87 @@
+package apimanagement
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/clients"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/apimanagement/parse"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/apimanagement/schemaz"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/tf/pluginsdk"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/timeouts"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func dataSourceApiManagementGateway() *pluginsdk.Resource {
+	return &pluginsdk.Resource{
+		Read: dataSourceApiManagementGatewayRead,
+
+		Timeouts: &pluginsdk.ResourceTimeout{
+			Read: pluginsdk.DefaultTimeout(5 * time.Minute),
+		},
+
+		Schema: map[string]*pluginsdk.Schema{
+			"name": schemaz.SchemaApiManagementChildDataSourceName(),
+
+			"resource_group_name": azure.SchemaResourceGroupNameForDataSource(),
+
+			"api_management_name": schemaz.SchemaApiManagementDataSourceName(),
+
+			"description": {
+				Type:     pluginsdk.TypeString,
+				Computed: true,
+			},
+
+			"location": {
+				Type:     pluginsdk.TypeList,
+				Computed: true,
+				Elem: &pluginsdk.Resource{
+					Schema: map[string]*pluginsdk.Schema{
+						"name": {
+							Type:     pluginsdk.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
```

Comment:
Since this is just one field, can we move this to a top level attribute and call it something like `location_name`? I don't think we should use `location` since that typically refers to an Azure location and they could be easily confused.

Reply:
Looking at the SDK, there are some other fields `City`, `District` and `CountryOrRegion` in addition to `Name` - do you know if there are populated and we should also include them?

Reply:
I've left it structured and included `City`, `District`, and `Region` fields. Can update if you feel would be worthwhile.